diesel_codegen_shared/src/schema_inference/pg.rs

```diff
@@ -99,14 +110,12 @@ pub fn get_primary_keys(conn: &PgConnection, table_name: &str) -> QueryResult<Ve
 }
 
 pub fn load_table_names(connection: &PgConnection) -> Result<Vec<String>, Box<Error>> {
-    use diesel::expression::dsl::sql;
-
-    let query = select(sql::<types::VarChar>("table_name FROM information_schema.tables"))
-        .filter(sql::<types::Bool>("\
-            table_schema = 'public' AND \
-            table_name NOT LIKE '\\_\\_%' AND \
-            table_type LIKE 'BASE TABLE'\
-        "));
+    use self::information_schema::tables::dsl::*;
+
+    let query = tables.select(table_name)
+        .filter(table_schema.eq("public"))
+        .filter(table_name.not_like("\\_\\_%"))
```

Comment:
Good to know! Might still read better as a raw string.

@weiznich I'm not sure I would expect diesel to escape this if you explicitly use `fn not_link(String)`. I found a method for stuff like `name.not_like("%a%%").escape('a')` though ([docs](http://docs.diesel.rs/diesel/expression/expression_methods/escape_expression_methods/trait.EscapeExpressionMethods.html)).